pkg/crc/preflight/preflight_linux.go

```diff
@@ -227,17 +227,15 @@ func getNetworkChecksForDistro(distro *linux.OsRelease, networkMode network.Mode
 		return append(checks, daemonRunningChecks)
 	}
 
-	switch {
-	default:
+	if !(distroIsLike(distro, linux.Ubuntu) || distroIsLike(distro, linux.Fedora)) {
 		logging.Warnf("distribution-specific preflight checks are not implemented for '%s'", distro.ID)
-		fallthrough
-	case distroIsLike(distro, linux.Ubuntu), distroIsLike(distro, linux.Fedora):
-		checks = append(checks, nmPreflightChecks[:]...)
-		if usesSystemdResolved(distro) {
-			checks = append(checks, systemdResolvedPreflightChecks[:]...)
-		} else {
-			checks = append(checks, dnsmasqPreflightChecks[:]...)
-		}
+	}
+
+	checks = append(checks, nmPreflightChecks[:]...)
+	if checkSystemdResolvedIsRunning() == nil {
+		checks = append(checks, systemdResolvedPreflightChecks[:]...)
+	} else {
+		checks = append(checks, dnsmasqPreflightChecks[:]...)
```

Comment:
Shouldn't this part of `else` block ? we don't want to have any nm preflight checks if distro is not like ubuntu or fedora since it's not tested and fix might not work.

Reply:
> fix might not work.
Are you referring to this patch or to the NetworkManager preflight operations? 

When authoring this, I was under the strict interpretation that NetworkManager is required, so if it were missing the run should fail.

From the docs:
> CodeReady Containers requires the libvirt and NetworkManager packages to run on Linux.

Are you meaning to say that the preflights may not function correctly on another distribution using NetworkManager due to that distribution’s configuration?

Reply:
@omenos Ignore it, I just missed we were using `fallthrough` so the logic is still the same.